blog/enable-cilium/index.mdx

```diff
@@ -0,0 +1,477 @@
+index 8d17418..5391776 100644
+--- a/edgecore.yaml
++++ b/edgecore.yaml
+@@ -62,6 +62,8 @@ modules:
+       cgroupDriver: systemd
+       cgroupsPerQOS: true
+       clusterDomain: cluster.local
++      clusterDNS:
++      - 10.96.0.10
+       configMapAndSecretChangeDetectionStrategy: Get
+       containerLogMaxFiles: 5
+       containerLogMaxSize: 10Mi
+@@ -151,7 +151,7 @@ modules:
+     enable: true
+     metaServer:
+       apiAudiences: null
+-      enable: false
++      enable: true
+       server: 127.0.0.1:10550
+       serviceAccountIssuers:
+       - https://kubernetes.default.svc.cluster.local
+@@ -161,7 +161,7 @@ modules:
+       tlsPrivateKeyFile: /etc/kubeedge/certs/server.key
+     remoteQueryTimeout: 60
+   serviceBus:
+-    enable: false
++    enable: true
+     port: 9060
+     server: 127.0.0.1
+     timeout: 60
+```
+
+And then, we need to create another `DaemonSet` of `cilium-agent` only for `EdgeCore` nodes.
+We need to deploy `cilium-agent` pods to the nodes where KubeEdge `EdgeCore` runs and labeled with `kubernetes.io/edge=`.
+Besides that, `cilium-agent` needs to query the APIs to `MetaServer` instead of Kubernetes API-server that is required to keep the edge autonomy provided by KubeEdge.
+
+```
+### Dump original Cilium DaemonSet configuration
+> kubectl get ds -n kube-system cilium -o yaml > cilium-edgecore.yaml
+
+### Edit and apply the following patch
+> vi cilium-edgecore.yaml
+```
+
+```patch
+diff --git a/cilium-edgecore.yaml b/cilium-edgecore.yaml
+index bff0f0b..3d941d1 100644
+--- a/cilium-edgecore.yaml
++++ b/cilium-edgecore.yaml
+@@ -8,7 +8,7 @@ metadata:
+     app.kubernetes.io/name: cilium-agent
+     app.kubernetes.io/part-of: cilium
+     k8s-app: cilium
+-  name: cilium
++  name: cilium-kubeedge
+   namespace: kube-system
+ spec:
+   revisionHistoryLimit: 10
+@@ -29,6 +29,12 @@ spec:
+         k8s-app: cilium
+     spec:
+       affinity:
++        nodeAffinity:
++          requiredDuringSchedulingIgnoredDuringExecution:
++            nodeSelectorTerms:
++              - matchExpressions:
++                - key: node-role.kubernetes.io/edge
++                  operator: Exists
+         podAntiAffinity:
+           requiredDuringSchedulingIgnoredDuringExecution:
+           - labelSelector:
+@@ -39,6 +45,8 @@ spec:
+       containers:
+       - args:
+         - --config-dir=/tmp/cilium/config-map
++        - --k8s-api-server=127.0.0.1:10550
++        - --auto-create-cilium-node-resource=true
+         - --debug
+         command:
+         - cilium-agent
+@@ -178,7 +186,9 @@ spec:
+       dnsPolicy: ClusterFirst
+       hostNetwork: true
+       initContainers:
+-      - command:
++      - args:
++        - --k8s-api-server=127.0.0.1:10550
++        command:
+         - cilium
+         - build-config
+         env:
+```
+
+We can see below that `cilium-pq45v` (normal cilium-agent pod) running on cloud node, and `cilium-kubeedge-kkb7z` (edgecore specific `DaemonSet`) is running with edgecore.
```

Comment:
should apply cilium-edgecore.yaml file `kubectl apply -f cilium-edgecore.yaml`.

Reply:
that is true, good eye! thanks!